src/citrine/informatics/executions/base_execution.py

```diff
@@ -0,0 +1,62 @@
+from abc import ABC
+from typing import List, Optional
+from uuid import UUID
+
+from citrine._rest.asynchronous_object import AsynchronousObject
+from citrine._rest.pageable import Pageable
+from citrine._rest.paginator import Paginator
+from citrine._serialization import properties
+from citrine._session import Session
+from citrine.resources.status_detail import StatusDetail
+
+
+from deprecation import deprecated
+
+
+class Execution(Pageable, AsynchronousObject, ABC):
```

Comment:
Added a base execution class. This is used for `design_execution`, `generative_design_execution`, `predictor_evaluation_execution` (this one doesn't NEED pagable, but also doesn't hurt?) and `sample_design_space_execution`.

Reply:
Also added `EXECUTING` to the `in_progress_statuses`, since we used that for the sample_design_space on the backend.

Reply:
I named the file `base_execution.py`. I'm wondering if that should just be called `Execution`. I was on the fence. (maybe there is a different better name).

Reply:
And yeah, I think just calling the file `execution.py` would be fine.